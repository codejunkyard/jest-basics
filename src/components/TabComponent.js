import React from "react";
import PropTypes from "prop-types";
import { Tabs, Tab } from "react-bootstrap";

export default class TabComponent extends React.PureComponent {
  static propTypes = {
    children: PropTypes.oneOfType([
      PropTypes.arrayOf(PropTypes.element),
      PropTypes.element
    ]).isRequired,
    title: (props, propName, componentName) => {
      console.log(props);

      props &&
        props.children &&
        props.children.forEach(child => {
          if (!child.props.title) {
            throw new Error(
              `The prop \`${propName}\` is marked as required in \`${componentName}.children()\`, but its value is \`${
                child.props.title
              }\`.`
            );
          }
        });
    }
  };

  state = {
    activeTab: 0
  };

  selectTab = tabId => {
    this.setState({ activeTab: tabId });
  };

  render() {
    const { children } = this.props;
    const { activeTab } = this.state;

    return (
      <Tabs activeKey={activeTab} onSelect={this.selectTab}>
        {children.map((child, index) => (
          <Tab key={index} eventKey={index} title={child.props.title}>
            {child}
          </Tab>
        ))}
      </Tabs>
    );
  }
}
