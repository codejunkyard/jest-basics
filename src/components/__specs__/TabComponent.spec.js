import React from "react";
import renderer from "react-test-renderer";
import { shallow, mount, render } from "enzyme";

import TabComponent from "../TabComponent";

describe("TabComponent", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <TabComponent>
          <p title="Test">Test</p>
          <p title="Test">Test</p>
        </TabComponent>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  // it("should shallow correctly", () => {
  //   expect(
  //     shallow(
  //       <TabComponent>
  //         <p title="Test">Test</p>
  //       </TabComponent>
  //     )
  //   ).toMatchSnapshot();
  // });

  // it("should mount correctly", () => {
  //   expect(
  //     mount(
  //       <TabComponent>
  //         <p title="Test">Test</p>
  //       </TabComponent>
  //     )
  //   ).toMatchSnapshot();
  // });

  // it("should render correctly", () => {
  //   expect(
  //     render(
  //       <TabComponent>
  //         <p title="Test">Test</p>
  //       </TabComponent>
  //     )
  //   ).toMatchSnapshot();
  // });
});
