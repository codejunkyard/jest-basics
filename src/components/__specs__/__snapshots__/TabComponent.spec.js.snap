// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabComponent renders correctly 1`] = `
Array [
  <nav
    className="nav nav-tabs"
    onKeyDown={[Function]}
    role="tablist"
  >
    <a
      aria-controls={null}
      aria-selected={true}
      className="nav-item nav-link active"
      data-rb-event-key="0"
      href="#"
      id={null}
      onClick={[Function]}
      onKeyDown={[Function]}
      role="tab"
    >
      Test
    </a>
    <a
      aria-controls={null}
      aria-selected={false}
      className="nav-item nav-link"
      data-rb-event-key="1"
      href="#"
      id={null}
      onClick={[Function]}
      onKeyDown={[Function]}
      role="tab"
      tabIndex={-1}
    >
      Test
    </a>
  </nav>,
  <div
    className="tab-content"
  >
    <div
      aria-hidden={false}
      aria-labelledby={null}
      className="fade tab-pane active show"
      id={null}
      role="tabpanel"
    >
      <p
        title="Test"
      >
        Test
      </p>
    </div>
    <div
      aria-hidden={true}
      aria-labelledby={null}
      className="fade tab-pane"
      id={null}
      role="tabpanel"
    >
      <p
        title="Test"
      >
        Test
      </p>
    </div>
  </div>,
]
`;
